experiments/AMIP/moist_mpi_earth/coupler_utils/bcfile_reader.jl

```diff
@@ -53,41 +56,35 @@ function bcfile_info_init(
     mono = true,
 )
 
-    # regrid all times and save to file
-    weightfile, datafile_cgll = ncreader_rll_to_cgll_from_space(
-        datafile_rll,
-        varname,
-        boundary_space,
-        outfile = varname * "_cgll.g",
-        mono = mono,
-    )
-    ds = Dataset(datafile_cgll, "r")
+    # regrid all times and save to hdf5 files
+    hd_outfile_root = varname * "_cgll"
+    if !is_distributed || ClimaComms.iamroot(comms_ctx)
```

Comment:
This should work generally:
```suggestion
    if ClimaComms.iamroot(comms_ctx)
```